### PR TITLE
LIC-917

### PR DESCRIPTION
**Pull request by yasinmustafa**

### CODE REVIEW

File: licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/pdf/PdfSpec.groovy [hunk 1]
```diff
@@ -77,10 +77,14 @@ class PdfSpec extends GebReportingSpec {
     to TaskListPage, testData.markAndrewsBookingId
     taskListAction('Create licence').click()
 
+    and: 'I choose a offence bfore'
```
Comment (bell-pa) [inline licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/pdf/PdfSpec.groovy]: Typo 'bfore'

File: server/routes/pdf.js [hunk 2]
```diff
@@ -39,6 +39,115 @@ module.exports = ({ pdfService, prisonerService }) => (router, audited) => {
...
 
+  router.get(
+    '/selectLicenceType/:bookingId',
+    asyncMiddleware(async (req, res) => {
+      const { bookingId } = req.params
+
+      const prisoner = await prisonerService.getPrisonerPersonalDetails(bookingId, res.locals.token)
+      let errors = firstItem(req.flash('errors')) || {}
+
+      let licenceTemplateId = getIn(res.locals.licence, ['approvedVersionDetails', 'template'])
```
Comment (bell-pa) [inline server/routes/pdf.js]: const

File: server/routes/pdf.js [hunk 3]
```diff
@@ -39,6 +39,115 @@ module.exports = ({ pdfService, prisonerService }) => (router, audited) => {
...
+        licenceTemplateId = ''
+      } else if (Object.keys(errors).includes('licenceTypeRadioListNo')) {
+        offenceBeforeCutoff = 'No'
+        const errorMessage = errors.licenceTypeRadioListNo
+        errors = { licenceTypeRadioList: errorMessage }
+        licenceTemplateId = ''
+      } else if (offenceCommittedBeforeFeb2015 === 'Yes' || offenceCommittedBeforeFeb2015 === 'No') {
+        offenceBeforeCutoff = offenceCommittedBeforeFeb2015
+      }
+
```
Comment (bell-pa) [inline server/routes/pdf.js]: Better to extract a function and use return values to initialise a const.

File: server/routes/pdf.js [hunk 4]
```diff
@@ -39,6 +39,115 @@ module.exports = ({ pdfService, prisonerService }) => (router, audited) => {
...
+    if (beforeLicenceTypeNotSelected(offenceBeforeCutoff, licenceTypeRadio)) {
+      req.flash('errors', { licenceTypeRadioListYes: 'Select a licence type' })
+      return res.redirect(`/hdc/pdf/selectLicenceType/${bookingId}`)
+    }
+    if (afterLicenceTypeNotSelected(offenceBeforeCutoff, licenceTypeRadio)) {
+      req.flash('errors', { licenceTypeRadioListNo: 'Select a licence type' })
+      return res.redirect(`/hdc/pdf/selectLicenceType/${bookingId}`)
+    }
+
+    const licenceTemplate = licenceTypeRadio
```
Comment (bell-pa) [inline server/routes/pdf.js]: Where is licenceTemplate used?

File: licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/pdf/PdfSpec.groovy (hunk 1 above)
Comment (yasinmustafa) [reply licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/pdf/PdfSpec.groovy]: Done

File: server/routes/pdf.js (hunk 2 above)
Comment (yasinmustafa) [reply server/routes/pdf.js]: Done

File: server/routes/pdf.js (hunk 3 above)
Comment (yasinmustafa) [reply server/routes/pdf.js]: Done

**Pull request merged**